internal/apis/certmanager/validation/certificate.go

```diff
@@ -118,6 +123,26 @@ func ValidateCertificateSpec(crt *internalcmapi.CertificateSpec, fldPath *field.
 		el = append(el, validateEmailAddresses(crt, fldPath)...)
 	}
 
+	if len(crt.OtherNames) > 0 {
+		if !utilfeature.DefaultFeatureGate.Enabled(feature.OtherNames) {
+			el = append(el, field.Forbidden(fldPath.Child("OtherNames"), "Feature gate OtherNames must be enabled on both webhook and controller to use the alpha `otherNames` field"))
+		}
+
+		for i, otherName := range crt.OtherNames {
+			if otherName.OID == "" {
+				el = append(el, field.Required(fldPath.Child("otherNames").Index(i).Child("oid"), "must be specified"))
+			}
+
+			if _, err := pki.ParseObjectIdentifier(otherName.OID); err != nil {
+				el = append(el, field.Invalid(fldPath.Child("otherNames").Index(i).Child("oid"), otherName.OID, "oid syntax invalid"))
+			}
+
+			if otherName.UTF8Value == "" {
+				el = append(el, field.Required(fldPath.Child("otherNames").Index(i).Child("utf8Value"), "must be specified"))
+			}
```

Comment:
suggestion (non-blocking): I know we say that we don't do any validation, but is there maybe something we can do? E.g. https://pkg.go.dev/unicode/utf8#ValidString ? Not a blocker, but just throwing the idea out there.